src/inputs/internal/SelectFieldBase.tsx

```diff
@@ -31,27 +39,87 @@ export interface SelectFieldBaseProps<O, V extends Value> extends BeamSelectFiel
  */
 export function SelectFieldBase<O, V extends Value>(props: SelectFieldBaseProps<O, V>): JSX.Element {
   const {
-    getOptionValue,
-    getOptionLabel,
-    getOptionMenuLabel = getOptionLabel,
+    compact = false,
+    disabled: isDisabled = false,
+    errorMsg,
+    helperText,
+    label,
+    readOnly: isReadOnly = false,
     onSelect,
+    fieldDecoration,
     options,
-    values,
+    onBlur,
+    onFocus,
+    inlineLabel,
     multiselect = false,
-    ...beamSelectFieldBaseProps
+    getOptionLabel,
+    getOptionValue,
+    getOptionMenuLabel = getOptionLabel,
+    sizeToContent = false,
+    values,
+    ...otherProps
   } = props;
 
-  const { contains } = useFilter({ sensitivity: "base" });
+  function onSelectionChange(keys: Selection): void {
```

Comment:
See red code below, but this is all "the same code", but just as a `function` declared in `SelectFieldBase` and used here directly, instead of passed as a prop to the now-inlined `ComboBox`.